bluesky/callbacks/core.py

```diff
@@ -686,13 +686,11 @@ def descriptor(self, doc):
         """Write the header for the actual scan data"""
         # List all scalar fields, excluding the motor (x variable).
         self._read_fields = sorted([k for k, v in doc['data_keys'].items()
-                                    if (v['object_name'] != self._motor
```

Comment:
Why this change? I think `object_name` is correct because the motor, extracted `doc['motors']` in Line 657 above, is an object name, not a data key. We want to filter out any data keys associated with `motor` (e.g., `motor_user_readback`, `motor_user_setpoint`).